docs/reference/contributing/target/target.md

```diff
@@ -34,9 +70,173 @@ Add the target description to `mbed-os\targets\targets.json` using keys that the
 }
 ```
 
-#### HAL porting
+### Directory structure
+
+The target’s directory structure has the following hierarchy: 
+
+- The top level lists the manufacturer. This is the same for all targets that a vendor supports.
+- The device family. This is specific to a family.
+- A specific device in the family. This only applies to the single device.
+- The specific board that uses the MCU in the previous level. You can use this for pin maps.
+
+```
+/mbed-os/targets/TARGET_<Manf>/TARGET_<Device_Family>/TARGET_<specific_MCU>/TARGET_<specific_board>/
+```
+
+This sample implementation uses:
+
+```
+/mbed-os/targets/TARGET_MY_VENDOR/TARGET_MY_FAMILY/TARGET_MY_DEVICE_1/TARGET_MY_BOARD_1/
+```
+
+### Entry in `mbed_rtx.h`
+
+The `mbed_rtx` header file defines the core requirements for the RTOS, such as clock frequency, initial stack pointer, stack size, task counts and other macros. This sample target has the following specifications:
+
+```
+
+    #elif defined(TARGET_MyTarget123)
+
+    #ifndef INITIAL_SP
+    #define INITIAL_SP              (0x20030000UL)
+    #endif
+
+    #if defined(__CC_ARM) || defined(__GNUC__)
+    #define ISR_STACK_SIZE          (0x1000)
+    #endif
+
+    #ifndef OS_TASKCNT
+    #define OS_TASKCNT              14
+    #endif
+    #ifndef OS_MAINSTKSIZE
+    #define OS_MAINSTKSIZE          256
+    #endif
+    #ifndef OS_CLOCK
+    #define OS_CLOCK                120000000
+    #endif   
+```
+
+Please refer to your chosen MCU's reference manual for these values.
+
+### Startup and HAL
+
+Ensure that the drivers are present at the correct level in the directory structure:
+
+1. Add startup code and CMSIS specific headers. You may obtain the startup code and other CMSIS specific headers from the device manufacturer or the CMSIS packs from KEIL&reg;.
+1. Add peripherals to the `device_has_add` key in `targets.json` and include all relevant drivers for all these peripherals.
+1. Implement Mbed OS HAL APIs for all hardware peripherals mentioned in `device_has`.
+1. Ensure that `Objects.h` declares peripherals and is available in the `/api` directory.
+
+### Linker files
+
+Each supported toolchain listed in your new target's `supported_toolchains` configuration needs a linker file. Each compiler supports a different style of linker files with a different extension:
+
+- Add a scatter file, with the extension `.sct`, for Arm Compiler 5.
+- Add a linker script, with the extension `.ld`, for GCC Arm Embedded.
+- Add an IAR Linker file, with the extension `.icf`, for IAR EWARM.
+
```

Comment:
@AnotherButler : Can you please add: "You will also need to add the startup codes (*.s) in the respective directories as well for ARMCC, GCC and IAR."

Reply:
@ashok-rao There is no reason that startup code has to be assembly, and that startup code cannot be shared between compilers. Assembly must have a capitol `.S` suffix.